config/lib.json

```diff
@@ -36,9 +36,9 @@
     "brotli": {
         "source": "brotli",
         "static-libs-unix": [
+            "libbrotlicommon.a",
             "libbrotlidec.a",
-            "libbrotlienc.a",
-            "libbrotlicommon.a"
+            "libbrotlienc.a"
```

Comment:
This is backwards now. Common must come last. SPCConfigUtil must use the order listed here before.

Reply:
I would use this to let pkg-config automatically order:

```json
{
    "brotli": {
        "source": "brotli",
        "pkg-configs": [
            "libbrotlicommon",
            "libbrotlidec",
            "libbrotlienc"
        ],
        "static-libs-windows": [
            "brotlicommon.lib",
            "brotlienc.lib",
            "brotlidec.lib"
        ],
        "headers": [
            "brotli"
        ]
    }
}
```

Reply:
Oh and last but not least, to improve linker performance, when we have a string like `-lssl -lcrypto -lz -lbrotlienc -lbrotlicommon -lbrotlienc -lbrotlicommon -lz` it should be deduplicated into -lssl -lycrypto -lbrotlienc -lbrotlidec -lbrotlicommon -lz` (only keep the last occurence of each library). It's not necessary, but can improve performance a lot.

Reply:
Of course we have: https://github.com/crazywhalecc/static-php-cli/blob/a095d489f35e6d2551ae45aed537708a9d5b36bc/src/SPC/util/SPCConfigUtil.php#L182-L183